src/test/java/teammates/test/pageobjects/InstructorCourseStudentDetailsEditPage.java

```diff
@@ -73,10 +73,19 @@ private void fillStudentDetailsForm(String studentName, String teamName, String
             fillTextBox(commentsTextbox, comments);
         }
         if (teamName == null) {
-            click(submitButton);
+            if (studentEmail == null) {
+                click(submitButton);
+            } else {
+                clickAndConfirm(submitButton);
+            }
+            
         } else {
             // if team name is edited, the confirmation dialog will pop up
-            clickAndConfirm(submitButton);
+            if (studentEmail == null) {
+                clickAndConfirm(submitButton);
+            } else {
+                clickAndConfirmTwoTimes(submitButton);
```

Comment:
Rather than doing this, why not call `clickAndConfirm` similarly and handle the second modal separately? That way you can simplify the routine:
```java
clickAndConfirm(submitButton);
if (studentEmail != null) {
    clickCancelOnModal(); // implement accordingly
}
```